sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexAsyncClient.java

```diff
@@ -542,6 +542,15 @@ SearchClientBuilder getSearchClientBuilder(String indexName) {
         }
     }
 
+    /**
+     * Lists all synonym map names for an Azure Cognitive Search service.
+     *
+     * @return a reactive response emitting the list of synonym map names.
+     */
+    public PagedFlux<String> listSynonymMapNames() {
```

Comment:
All APIs in this client are missing @ServiceMethod annotation. Also, add codesnippets where appropriate.

Reply:
I will add @ServiceMethod. 

Can improve JavaDoc and other documents in preview 5.